### PR TITLE
Count closures arguments for the NArgs metric

### LINKED ISSUE #342: Consider closures arguments for Nargs metric

### PR DESCRIPTION
**Pull request by Luni-4**
Depends on #373 

This PR fixes #342 

Thanks in advance for your review! :)

### CODE REVIEW

File: src/output/dump_metrics.rs [hunk 1]
```diff
@@ -99,8 +99,8 @@ fn dump_metrics(
     let prefix = format!("{}{}", prefix, pref_child);
     dump_cognitive(&metrics.cognitive, &prefix, false, stdout)?;
     dump_cyclomatic(&metrics.cyclomatic, &prefix, false, stdout)?;
-    dump_nargs(&metrics.nargs, &prefix, false, stdout)?;
     dump_nexits(&metrics.nexits, &prefix, false, stdout)?;
+    dump_nargs(&metrics.nargs, &prefix, false, stdout)?;
```
Comment (marco-c) [inline src/output/dump_metrics.rs:103]: Why did you move it down?

File: src/metrics/fn_args.rs [hunk 2]
```diff
@@ -1,196 +0,0 @@
-use serde::ser::{SerializeStruct, Serializer};
```
[the remark this answers is not in the document]
Comment (Luni-4) [reply src/metrics/fn_args.rs:1]: It's part of this PR, because we are not counting only `fn_args` anymore, but also `closures_args`

File: rust-code-analysis-cli/src/web/server.rs [hunk 3]
```diff
@@ -633,7 +633,7 @@ mod tests {
                        "end_line": 4,
                        "metrics": {"cyclomatic": {"sum": 2.0, "average": 1.0},
                                    "cognitive": {"sum": 0.0, "average": 0.0},
-                                   "nargs": {"sum": 0.0, "average": 0.0},
+                                   "nargs": {"functions": 0.0, "closures": 0.0, "total": 0.0, "average": 0.0},
```
Comment (marco-c) [inline rust-code-analysis-cli/src/web/server.rs:636]: Maybe we should have total, average, total_functions, average_functions, total_closures, average_closures

File: src/metrics/fn_args.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (marco-c) [reply src/metrics/fn_args.rs:1]: OK, but please do it in a separate commit, otherwise reviewing is hard.

[the remark this answers is not in the document]
Comment (Luni-4) [reply src/metrics/fn_args.rs:1]: Sure :)

File: rust-code-analysis-cli/src/web/server.rs (hunk 3 above)
Comment (Luni-4) [reply rust-code-analysis-cli/src/web/server.rs:636]: I can add them

**Comment by Luni-4** [comment]
Depends on #373

File: src/metrics/nargs.rs [hunk 4]
```diff
@@ -412,7 +412,7 @@ mod tests {
     #[test]
     fn javascript_single_closure() {
         check_metrics!(
-            "var bar = function (a, b) {return a + b};",
```
Comment (marco-c) [inline src/metrics/nargs.rs]: Can you split this into two tests instead of modifying it?

File: src/metrics/nargs.rs [hunk 5]
```diff
@@ -713,26 +713,26 @@ mod tests {
     fn javascript_nested_functions() {
         check_metrics!(
             "function f(a, b) {
```
Comment (marco-c) [inline src/metrics/nargs.rs:715]: Same here, can you split this into two tests instead of modifying it?

File: src/metrics/nargs.rs [hunk 6]
```diff
@@ -8,18 +8,22 @@ use crate::*;
 /// The `NArgs` metric.
 ///
 /// This metric counts the number of arguments
-/// of a function/method.
+/// of a function/closures.
```
Comment (marco-c) [inline src/metrics/nargs.rs:11]: `of a function/closure` or `of functions/closures`

**Review by marco-c — DISMISSED** [review]
Thanks!

**Review by marco-c — APPROVED** [review]
